### PR TITLE
Adding support for custom header menu

### PR DESCRIPTION
**Pull request by ahmad-shahid**

### CODE REVIEW

File: Components/IModel.cs [hunk 1]
```diff
@@ -98,6 +98,12 @@ public interface IModel
         /// </summary>
         List<FooterSection> FooterSections { get; set; }
 
+        /// <summary>
+        /// Used to display a custom header menu on the top of the page for app templates
+        /// Set by application programmatically
```
Comment (plup2) [inline Components/IModel.cs:103]: We should probably add a mention similar to FooterSections above that it is only available in the application template for GGCweb theme.  (I forgot to mention that for the banner :) but these comments can end up as documentation for users (at least in Java they do))

Comment (ahmad-shahid) [reply Components/IModel.cs:103]: Should be a separate issue looool jk jk I'll make the change :)

Comment (ahmad-shahid) [reply Components/IModel.cs:103]: done!

Comment (plup2) [reply Components/IModel.cs:103]: :)

**Review by plup2 — APPROVED** [review]

**Pull request merged**